src/module/roll-table/roll-table.ts

```diff
@@ -0,0 +1,358 @@
+import type { ChatSpeakerData } from '@league-of-foundry-developers/foundry-vtt-types/src/foundry/common/data/data.mjs/chatSpeakerData'
+import type { RollTableDataConstructorData } from '@league-of-foundry-developers/foundry-vtt-types/src/foundry/common/data/data.mjs/rollTableData'
+import type { ConfiguredFlags } from '@league-of-foundry-developers/foundry-vtt-types/src/types/helperTypes'
+import type { IOracle, IRow } from 'dataforged'
+import { max } from 'lodash-es'
+import { marked } from 'marked'
+import type { IronswornActor } from '../actor/actor'
+import { hashLookup, renderLinksInStr } from '../dataforged'
+import {
+	findPathToNodeByTableUuid,
+	getOracleTreeWithCustomOracles
+} from '../features/customoracles'
+import { cachedDocumentsForPack } from '../features/pack-cache'
+
+import { OracleTableResult } from './table-result'
+import type { ComputedTableType } from './roll-table-types'
+
+/** Extends FVTT's default RollTable with functionality specific to this system. */
+export class OracleTable extends RollTable {
```

Comment:
Probably this file should be called `oracle-table.ts` because of the class it's mainly here for.

Reply:
funnily enough, originally it was named that, but later opted to follow the pattern set in the actor/item class file names (e.g. `actor.ts` for `IronswornActor`). i'll go and change it back -- probably clearer that way